lib/transports/TCP.js

```diff
@@ -247,4 +247,102 @@ Transport.prototype.close = function () {
     this.connection.destroy();
     this.closed = true;
     this.emit('close');
+};
+
+
+function ClientTransport(config){
+    var self = this;
+    
+    self.host = config.host;
+    self.port = config.port;
+    self.heloFromClient = config.heloFromClient;
+    self.heloFromServer = config.heloFromServer;
+
+    self.closed = false;
+    self.connection = new net.Socket();
+
+    self.connection.on('data', function (data) {
+        self.emit('data', data);
+    });
+
+    self.connection.on('close', function () {
+        console.log('c -> rcvd close');
+        if (!self.closed) {
+            self.closed = true;
+            self.emit('close');
+        }
+    });
+}
+util.inherits(ClientTransport,ClientTransportPrototype);
+
+ClientTransport.prototype.connect = function(cid) {
+    
+    //TODO add cid handler
+    var self = this;
+    
+    var data = new Buffer(0);
+    var gotHelo = false;
+    var helo_bl = Buffer.byteLength(self.heloFromServer);
+    var CIDMarker = null;
+    var gotCIDMarker = false;
+    var gotCID = false;
+
+    self.connection.connect(self.port, self.host);        
+    
+    self.connection.write(self.heloFromClient);
+    self.connection.write("#");    
+    
+    //wait hello from server
+    var onData = function(dataReceived) {        
+        data = Buffer.concat([data,dataReceived],data.length + dataReceived.length);        
+        if (!gotHelo && data.length < helo_bl) return;
+        // Wrong helo?        
+        if (!gotHelo && data.toString('utf8',0,helo_bl) != self.heloFromServer) {                        
+            self.connection.end("WRONG HELO");
+            return;
+        } 
+          
+        if(!gotHelo) {
+            // Если HELO подошел мы должны обрезать данные, чтобы передать их дальше.
+            data = data.slice(helo_bl);
+            gotHelo = true;
+        }     
+
+        if (!gotCIDMarker && data.length === 0) return;        
+        if(CIDMarker === null) {            
```

Comment:
Client transport should write CID marker, so server transport could read it. Now they both read it.